### PR TITLE
[Hub Generated] Review request for Microsoft.CostManagement to add version 2019-03-01-preview

### PR DESCRIPTION
**Pull request by ariehf**
Main changes:
- Field changes related to CloudConnector,ExternalBillingAccount,ExternalSubscription
- Remove API POST: /providers/Microsoft.Management/managementGroups/{managementGroupId}/providers/Microsoft.CostManagement/getEntities
- Reformat costmanagement.json with VisualStudio Code
- Fix typo Managment -> Management
- Fix validation errors in Showback

Details of Field changes:
- CloudConnector: renamed providerAccountId to providerBillingAccountId
- CloudConnector: added providerBillingAccountDisplayName, externalBillingAccountId
- CloudConnector: changed status to be readonly (no longer part of PUT)
- CloudConnector: changed PATCH to not take a connector properties, and therefore to be used only for verifying an existing cloudconnector
- ExternalBillingAccount: added providerBillingAccountId, connectorId
- ExternalBillingAccount: added GET $expand=collectionInfo
- ExternalSubscription: added providerBillingAccountId, providerAccountId
- ExternalSubscription: added GET $expand=collectionInfo
- ManagementGroups/externalSubscription: added $recurse

If you are a MSFT employee you can view your [work branch via this link](https://portal.azure-devex-tools.com/app/branch/ariehf/azure-rest-api-specs/dev-cost-management-Microsoft.CostManagement-2019-03-01-preview).

### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by praries880** [comment]
@AutorestCI regenerate azure-sdk-for-go

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json [hunk 1]
```diff
@@ -1,3215 +1,3214 @@
...
-					"description": "Connector type"
-				},
-				"properties": {
-					"x-ms-client-flatten": true,
-					"$ref": "#/definitions/ConnectorProperties",
-					"description": "Connector properties"
-				}
-			},
-			"x-ms-azure-resource": true
-		}
```
Comment (nschonni) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: Managment -> Management

Comment (nschonni) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: Managment -> Management

Comment (ariehf) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: Done

**Comment by praries880** [comment]
@AutorestCI regenerate azure-sdk-for-go

**Comment by praries880** [comment]
@sergey-shandar Can you look into the issue with the Defrault build issue?

https://azuresdkci.westus2.cloudapp.azure.com/job/DotNetSwaggerToSdkCI/73/

**Comment by dsgouda** [comment]
startbuild

**Comment by shahabhijeet** [comment]
startbuild

**Comment by shahabhijeet** [comment]
@praries880 ignore failures on "default" job.
We recently renamed the job to "Automation for azure-sdk-for-net"

**Comment by sanjaiganesh** [comment]
Arieh, There are lot of breaking changes and must be done in new api version.

**Comment by sanjaiganesh** [comment]
>         "operationId": "CloudConnector_Verify",

why renaming a patch operation as '_verify' , instead of '_update' ?

---
Refers to: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json:1326 in ef9b6d3. [](commit_id = ef9b6d3914b85107a9d26a999f30f44c3564395f, deletion_comment = False)

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/-1 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/-1/commits/

**Comment by ariehf** [comment]
> > ```
> >     "operationId": "CloudConnector_Verify",
> > ```
> 
> why renaming a patch operation as '_verify' , instead of '_update' ?
> 
> Refers to: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json:1326 in [ef9b6d3](https://github.com/Azure/azure-rest-api-specs/commit/ef9b6d3914b85107a9d26a999f30f44c3564395f). [](commit_id = [ef9b6d3](https://github.com/Azure/azure-rest-api-specs/commit/ef9b6d3914b85107a9d26a999f30f44c3564395f), deletion_comment = False)

Renamed _verify to _update

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/-1 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/-1/commits/

**Comment by praries880** [comment]
@sanjaiganesh / @antmarti-microsoft Can you validate that the latest commits handle the ARM feeback?

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json [hunk 2]
```diff
@@ -1,3215 +1,3193 @@
...
```
[the remark this answers is not in the document]
Comment (ariehf) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: RBAC checks are handled within the service.
A large customers, can have around 1000 External Subscriptions. The ARM batch limit is 20, which means 50 sequential batches.

Comment (antmarti-microsoft) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: What does "any of its descendants" mean here? Is there anyway for the customer to bypass RBAC with this API (i.e. fetching data they shouldn't have access to)?

Comment (antmarti-microsoft) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: Is this description + operation name accurate? It sounds more like it's the other way around - you're updating an external subscription for a management group.

Comment (antmarti-microsoft) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: This feels like a breaking change. Is this data ever returned from the server to the client, or is it only valid on a request? Will your service ever return "suspended", or accept "suspended" from an older version of the client going forwards?

Have you generated client SDKs for this swagger version yet?

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json [hunk 3]
```diff
@@ -1,3215 +1,3193 @@
...
-					"default": {
-						"description": "Error response describing why the operation failed.",
-						"schema": {
-							"$ref": "#/definitions/ErrorResponse"
-						}
-					}
-				}
-			}
-		},
-		"/providers/Microsoft.Management/managementGroups/{managementGroupId}/providers/Microsoft.CostManagement/getEntities": {
```
Comment (antmarti-microsoft) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: Removing this API from an existing API version is considered a breaking change. Have you verified that this URL is not handling any traffic? Have client SDKs been generated for this API version?

**Review by anthony-c-martin — CHANGES_REQUESTED** [review]

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json (hunk 3 above)
Comment (ariehf) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: This URL was never available in production. No client SDKs were generated with this API.

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json (hunk 2 above)
Comment (ariehf) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: This service is only currently available via a feature flag, and no generated client SDKs yet

Comment (ariehf) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: A management group can have many external subscriptions. An external subscription can have max one management group. The existing "ExternalSubscription_UpdateManagementGroup" seems correct

Comment (ariehf) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: With or without the $recurse parameter, this method only returns external subscriptions associated to management groups that they have suitable permissions for. However the $recurse parameter just indicates if the management group mentioned refers to only that MG or its descendants as well.

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json [hunk 4]
```diff
@@ -1,3215 +1,3193 @@
...
-					}
-				},
-				"parameters": [{
-						"$ref": "#/parameters/apiVersionParameter"
-					},
-					{
-						"$ref": "#/parameters/connectorNameParameter"
-					},
-					{
-						"name": "connector",
```
[the remark this answers is not in the document]
Comment (ariehf) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: Added it in

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (ariehf) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json]: Removed the multi-update operation

**Review by anthony-c-martin — APPROVED** [review]

**Review by praries880 — APPROVED** [review]

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/-1 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/-1/commits/

**Pull request merged**